opensrp-core/src/main/java/org/smartregister/util/OpenSRPImageLoader.java

```diff
@@ -493,118 +271,71 @@ public OpenSRPImageLoader setMaxImageSize(int maxImageSize) {
      * the name of the file is equals to the client's base entity id.
      *
      * @param entityId - The id of the image to be retrieved
-     * @return ImageContainer that will contain either the specified default bitmap or the loaded
-     * bitmap. If the default was returned, the
-     * {@link OpenSRPImageLoader} will be invoked when the request is fulfilled.
      */
     public void getImageByClientId(String entityId, OpenSRPImageListener opensrpImageListener) {
-
         try {
-            if (entityId == null || entityId.isEmpty()) {
+            final Context context = contextWeakReference.get();
+            if (context != null) {
+                if (CoreLibrary.getInstance().context().getAppProperties().isTrue(AllConstants.PROPERTY.DISABLE_PROFILE_IMAGES_FEATURE)
+                        || (entityId == null || entityId.isEmpty())) {
+
+                    Glide.with(context).load(opensrpImageListener.getDefaultImageResId()).into(opensrpImageListener.getImageView());
 
-                // If imageId is NULL, just return the image with resource id "defaultImageResId"
-                ImageContainer imgContainer = new ImageContainer(null, null, null,
-                        opensrpImageListener);
+                    return;
 
-                opensrpImageListener.onResponse(imgContainer, true);
-                return;
+                } else {
 
-            } else {
-                //get image record from the db
-                opensrpImageListener.setEntityId(entityId);
-                LoadProfileImageTask loadProfileImageTask = new LoadProfileImageTask(this,
-                        opensrpImageListener, entityId);
-                startAsyncTask(loadProfileImageTask, null);
+                    //To do in background
 
-            }
-        } catch (Exception e) {
-            Timber.e(e.getMessage(), e);
-        }
-    }
+                    ImageRepository imageRepo = CoreLibrary.getInstance().context().imageRepository();
+                    ProfileImage imageRecord = imageRepo.findByEntityId(entityId);
 
-    public void get(final ProfileImage image, final OpenSRPImageListener opensrpImageListener) {
+                    if (imageRecord != null) {
 
-        try {
-            // Non existent image record, display image with defaultImageResId
-            if (image == null) {
-                ImageContainer imgContainer = new ImageContainer(null, null, null,
-                        opensrpImageListener);
-                opensrpImageListener.onResponse(imgContainer, true);
-                return;
-            }
-            opensrpImageListener.setAbsoluteFileName(image.getFilepath());
+                        // Non existent image record, display image with defaultImageResId
+                        if (imageRecord == null) {

```

Comment:
This block should probably be in the `else` since the `imageRecord` cannot be null at this point

Reply:
You are right, we actually don't need it here since the call to configure the defaults is invoked in the else